### PR TITLE
Acceptance testing and test workflow

### PR DESCRIPTION
**Pull request by tylerjl**
Several changes here, in summary:

- I slightly tweaked the devshell setup - everything should be functionally the same as before, except that it's got more utilities inside of it and managed with a nix flake now. Should be effectively a no-op (nix-shell still works)
- Some of the docs were out-of-sync with the provider inline documentation - some resources lost their descriptors, but I have a separate branch where I'm working on beefing those up.
- There's now some task automation in `justfile` to to things like generate SITE_IDs, username/passowords, and stand up a test container. TODO here to make the version less static and maybe track our latest release.
- A new acceptance test for sites. This can be driven with `just acceptance-test`, which mostly bootstraps the container, runs go test with the right flags, and tears down the container again.
- I rolled this test suite into a workflow action to run tests on push to `main` and pull requests.
- Some minor linting checks that the golangci-linter caught; but it ended up complaining about some unused policy-related resources. I want to leave those in there for now while we work on them, so I just commented that out for now.

There's more work to be done to write more acceptance tests but I wanted to get this in as a first step - with actual tests in place we can start branching out into other work more easily.

### CODE REVIEW

File: internal/bowtie/client/http_test.go [hunk 1]
```diff
@@ -35,6 +35,21 @@ func TestClient_getHostURL(t *testing.T) {
...
 			want: "https://example.com/-net/api/v0/user/login",
 		},
+		{
+			name: "local login",
+			fields: fields{
+				HTTPClient: nil,
+				hostURL:    "http://localhost:3000",
+				auth: AuthPayload{
+					Username: "test@example.com",
+					Password: "passw0rd123",
```
Comment (issackelly) [inline internal/bowtie/client/http_test.go:45]: Does this purposefully fail zxcvbn? 
https://old.formvalidation.io/examples/building-password-strength-meter/

**Comment by issackelly** [comment]
This is a great set of changes. I could see the `justfile` becoming its own thing for other container testing (openapi fuzz tests anybody?) and I'm glad to see this moving forward here.

File: internal/bowtie/client/http_test.go (hunk 1 above)
Comment (chriskuchin) [reply internal/bowtie/client/http_test.go:45]: All this test is doing is testing that my url building template works for calculating where to send requests. My guess is tyler added this one to just make sure we caught any awkwardness with http and localhost

**Comment by chriskuchin** [comment]
Looks great to me Will sync over to my repo and use it to fix the dns read bugs.. The last patch seems to be slightly off

File: internal/bowtie/client/http_test.go (hunk 1 above)
Comment (tylerjl) [reply internal/bowtie/client/http_test.go:45]: Yeah, that's all I was doing here - ensuring that the login logic was hitting the endpoint I expected.

**Comment by tylerjl** [comment]
@chriskuchin one thing I almost forgot to mention is that, although I did add in the logic that additionally accepts `See Other` as a valid login response - which should still be backwards compatible - [this change](https://github.com/bowtieworks/terraform-provider-bowtie/pull/5/files#diff-a3aa89106deaf84a8ae8f142125e5cdf66217416ceb946f07fde171795585c68) was a result of our debugging but in hindsight probably wasn't the main issue (it was probably the cookies all along). Should I leave that in or take it out? I'm not a go expert but I assume it just continues to work the way it did with the addition that it's okay with redirects.

**Pull request merged**